nexus/src/app/mod.rs

```diff
@@ -248,9 +246,34 @@ impl Nexus {
             sec_store,
         ));
 
+        // It's a bit of a red flag to use an unbounded channel.
+        //
+        // This particular channel is used to send a Uuid from the saga executor
+        // to the saga recovery background task each time a saga is started.
+        //
+        // The usual argument for keeping a channel bounded is to ensure
+        // backpressure.  But we don't really want that here.  These items don't
+        // represent meaningful work for the saga recovery task, such that if it
+        // were somehow processing these slowly, we'd want to slow down the saga
+        // dispatch process.  Under normal conditions, we'd expect this queue to
+        // grow as we dispatch new sagas until the saga recovery task runs, at
+        // which point the queue will quickly be drained.  The only way this
+        // could really grow without bound is if the saga recovery task gets
+        // completely wedged and stops receiving these messages altogether.  In
+        // this case, the maximum size this queue could grow over time is the
+        // number of sagas we can launch in that time.  That's not ever likely
+        // to be a significant amount of memory.
+        //
+        // We could put our money where our mouth is: pick a sufficiently large
+        // bound and panic if we reach it.  But "sufficiently large" depends on
+        // the saga creation rate and the period of the saga recovery background
+        // task.  If someone changed the config, they'd have to remember to
+        // update this here.  This doesn't seem worth it.
```

Comment:
I will freely admit that i'm very much in the "unbounded channels should never be used, you should feel bad for using them" camp, but this comment has convinced me that it makes sense in this case. 

Thinking through it, i can't really see another approach having meaningfully different behavior WRT memory growth, anyway --- if we exerted backpressure on callers, then the Tokio tasks that are actually starting those sagas will probably stay in memory longer, because they're waiting on the queue to actually run, and a tokio task is, of course, much larger than a UUID. So, it's kind of a wash. You've sold me, thanks for writing it down!

Reply:
I am also very much in that camp, for what it's worth.  That's why I felt it needed an apology 😄